odysseus/odysseus_tree/board/tpu/plugin_user_property_timestamp.c

```diff
@@ -0,0 +1,48 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/time.h>
+
+
+#include "nanomq/plugin.h"
+
+/*
+ * How to compile:
+ * gcc -I../ -fPIC -shared plugin_user_property.c -o plugin_user_property.so
+ */
+
+
+int cb(void *data)
+{
+	char **property = data;
+	if (property != NULL) {
+		
+		// key of ts, add 1 bc termination byte
+		property[0] = malloc(strlen("ts") + 1);
```

Comment:
why do you need to malloc? shouldnt this get reset after each call

Reply:
Since the values need to outlive the function when being passed back up to the main system?

Reply:
malloc here feels funky, especially because we never free anything and nanomq doesn't know that there is memory to be freed. Is there not a buffer we can just copy stuff into with `memcpy`? or `strcpy`?

Reply:
Resolved in slack, NanoMQ frees the `data` pointer later on

Reply:
Feels like a weird way to do it but :shrug: